contracts/PartyBid.sol

```diff
@@ -0,0 +1,701 @@
+    uint256 public highestBid;
+    // the total spent by PartyBid on the auction;
+    // 0 if the NFT is lost; highest bid + 5% PartyDAO fee if NFT is won
+    uint256 public totalSpent;
+    // contributor => array of Contributions
+    mapping(address => Contribution[]) public contributions;
+    // contributor => total amount contributed
+    mapping(address => uint256) public totalContributed;
+    // contributor => true if contribution has been claimed
+    mapping(address => bool) public claimed;
+
+    // ============ Events ============
+
+    event Contributed(
+        address indexed contributor,
+        uint256 amount,
+        uint256 previousTotalContributedToParty,
+        uint256 totalFromContributor
+    );
+
+    event Bid(uint256 amount);
+
+    event Finalized(PartyStatus result, uint256 totalSpent, uint256 fee, uint256 totalContributed);
+
+    event Claimed(
+        address indexed contributor,
+        uint256 totalContributed,
+        uint256 excessContribution,
+        uint256 tokenAmount
+    );
+
+    // ======== Modifiers =========
+
+    modifier onlyPartyDAO() {
+        require(
+            msg.sender == partyDAOMultisig,
+            "PartyBid:: only PartyDAO multisig"
+        );
+        _;
+    }
+
+    // ======== Constructor =========
+
+    constructor(
+        address _partyDAOMultisig,
+        address _tokenVaultFactory,
+        address _weth
+    ) {
+        partyDAOMultisig = _partyDAOMultisig;
+        tokenVaultFactory = _tokenVaultFactory;
+        weth = _weth;
+    }
+
+    // ======== Initializer =========
+
+    function initialize(
+        address _marketWrapper,
+        address _nftContract,
+        uint256 _tokenId,
+        uint256 _auctionId,
+        address _partyHost,
+        uint256 _partyHostBasisPoints,
+        string memory _name,
+        string memory _symbol
+    ) external initializer {
+        // initialize ReentrancyGuard and ERC721Holder
+        __ReentrancyGuard_init();
+        __ERC721Holder_init();
+        // set storage variables
+        marketWrapper = _marketWrapper;
+        nftContract = _nftContract;
+        tokenId = _tokenId;
+        auctionId = _auctionId;
+        name = _name;
+        symbol = _symbol;
+        // validate that party host won't retain the total token supply
+        uint256 _remainingBasisPoints = 10000 - TOKEN_FEE_BASIS_POINTS;
+        require(_partyHostBasisPoints < _remainingBasisPoints, "PartyBid::initialize: basis points can't take 100%");
+        // validate that a portion of the token supply is not being burned
+        if (_partyHost == address(0)) {
+            require(_partyHostBasisPoints == 0, "PartyBid::initialize: can't send tokens to burn addr");
+        }
+        partyHostBasisPoints = _partyHostBasisPoints;
+        partyHost = _partyHost;
+        // validate token exists
+        require(_getOwner() != address(0), "PartyBid::initialize: NFT getOwner failed");
```

Comment:
generally speaking, if the owner is address(0) it means the token does not exist, and typically the ERC-721 contract will revert when calling `ownerOf` in this case. 
we wrote the `_getOwner` function so that `ownerOf` will not revert in this case, but rather just return address(0)